homeassistant/components/tplink/__init__.py

```diff
@@ -87,7 +111,40 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     host = entry.data[CONF_HOST]
     try:
-        device: SmartDevice = await Discover.discover_single(host, timeout=10)
+        credentials = await get_stored_credentials(hass)
+        connection_params = None
+
+        try:
+            if connection_params_dict := entry.data.get(CONF_CONNECTION_PARAMS):
+                connection_params = ConnectionParameters.from_dict(
+                    connection_params_dict
+                )
+        except SmartDeviceException:
+            _LOGGER.warning(
+                "Invalid connection parameters for %s: %s", host, connection_params_dict
+            )
+
+        device: SmartDevice = await Discover.connect_single(
```

Comment:
Can this raise?

Reply:
Yes, you've made me realise I should make the try block that is happening in less broad.

Reply:
Reduced the scope of the try block now